fe/fe-core/src/main/java/org/apache/doris/catalog/Env.java

```diff
@@ -5379,6 +5381,62 @@ public void cleanTrash(AdminCleanTrashStmt stmt) {
         }
     }
 
+    public void setPartitionVersion(AdminSetPartitionVersionStmt stmt) throws DdlException {
+        long partitionId = stmt.getPartitionId();
+        long versionInfo = stmt.getVisibleVersion();
+        int setSuccess = setPartitionVersionInternal(partitionId, versionInfo, false);
+        if (setSuccess == -1) {
+            throw new DdlException("Failed to set partition visible version to " + versionInfo + ". "
+                + "Partition " + partitionId + " not exists.");
+        }
+    }
+
+    public void replaySetPartitionVersion(SetPartitionVersionOperationLog log) throws DdlException {
+        int setSuccess = setPartitionVersionInternal(log.getPartitionId(), log.getVersionInfo(), true);
+        if (setSuccess == -1) {
+            LOG.warn("Failed to set partition visible version to {}. Partition {} not exists.",
+                    log.getVersionInfo(), log.getPartitionId());
+        }
+    }
+
+    public int setPartitionVersionInternal(long partitionId, long versionInfo, boolean isReplay) throws DdlException {
+        int result = -1;
+        List<Long> dbIds = getInternalCatalog().getDbIds();
+        // TODO should use inverted index
+        for (long dbId : dbIds) {
```

Comment:
how about set dbid and table id by user?
Iterate all dbs and tables is very expansive.

Reply:
Sure.